src/vos/vos_tree.c

```diff
@@ -1010,6 +1010,15 @@ key_tree_punch(struct vos_object *obj, daos_handle_t toh, daos_epoch_t epoch,
 	return rc;
 }
 
+int
+key_tree_delete(struct vos_object *obj, daos_handle_t toh, d_iov_t *key_iov)
+{
+	/* Delete dkey or akey
+	 * TODO: incarnation log should be deleted as well.
+	 */
+	return dbtree_delete(toh, BTR_PROBE_EQ, key_iov, NULL);
```

Comment:
ktr_rec_free destroys the incarnation log before adding the tree to the garbage collector so I don't think the comment is necessary